backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +55,64 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(
+    bind=True,
+    queue="indexer_task",
+    acks_late=True,
+    retry_backoff=True,
+    retry_kwargs={"max_retries": 3},
+    autoretry_for=(OperationalError, TransportError),
+)
+def redirect_references_to_merged_record(self, uuid):
+    record = InspireRecord.get_record(uuid)
+    new_record_ref = record["new_record"]["$ref"]
+    deleted_record_ref = record["self"]["$ref"]
+    record_schema = PidStoreBase.get_schema_name_from_uri(record["$schema"])
+    possible_refs_to_record = get_refs_to_schemas()[record_schema]
+    update_references_pointing_to_merged_record(
+        possible_refs_to_record, deleted_record_ref, new_record_ref
+    )
+
+
+def update_references_pointing_to_merged_record(
+    refs_to_schema, merged_record_uri, new_record_uri
+):
+    for index, path in refs_to_schema:
+        config = get_config_for_given_path(index, path)
+        matched_records = match({"$ref": merged_record_uri}, config)
+        for matched_record in matched_records:
+            matched_inspire_record = InspireRecord.get_record(matched_record["_id"])
+            referenced_records_in_path = flatten_list(
+                get_value(matched_inspire_record, path[: -len(".$ref")], [])
+            )
+            for referenced_record in force_list(referenced_records_in_path):
```

Comment:
I don't think you need the `force_list` here, as `flatten_list` already returns a `list`.

Reply:
For authors.affiliations it doesn't work, probably because it's an array in nested fields

Reply:
that's why you need `flatten_list`, but `force_list` on top doesn't add anything (unless there's a bug in the former).